### PR TITLE
feat: recommend hub's tasks based on user project (#430)

### LINKED ISSUE #430: TektonHub: recommend task based on user project
Alizer can return language for the project(s) so we could add an option for filter the result

### PR DESCRIPTION
**Pull request by lstocchi**
it resolves #430

### CODE REVIEW

**Comment by jeffmaury** [comment]
Please rebase, I'm afraid you'll have to revert your last commit because of 6d78d83b9535e390ad584be2546dea65ac0930e5

**Comment by lstocchi** [comment]
I already rebased it yesterday otherwise the build was broken

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/hub/HubModel.java [hunk 1]
```diff
@@ -83,8 +85,10 @@ public HubModel(Project project, Tkn tkn, String namespace, List<String> tasks,
     public List<HubItem> getAllHubItems() {
         if (allHubItems.isEmpty()) {
             try {
-                allHubItems = retrieveAllHubItems().get();
-            } catch (InterruptedException | ExecutionException e) {
+                List<Language> languages = new LanguageRecognizerBuilder().build().analyze(project.getBasePath());
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/hub/HubModel.java:88]: `Project.getBasePath()` is not working in all cases. Project in IntelliJ is a workspace that contains a collection of Modules (projects). So I suspect you can an empty project in directory A and add a Maven or Gradle module from directory B.
But the current implementation will work for 80% of the user cases, I think.

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Also tried to test it but never got the Hub dialog being shown: I selected the Import from Hub on the Tasks node and got this exception being logged:

```
update failed for AnAction(com.redhat.devtools.intellij.common.actions.editor.RemoveYAMLClutterAction) with ID=com.redhat.devtools.intellij.common.actions.editor.RemoveYAMLClutterAction

java.lang.NullPointerException
	at com.redhat.devtools.intellij.common.actions.editor.YAMLClutterAction.update(YAMLClutterAction.java:43)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:177)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doUpdate(ActionUpdater.java:444)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$null$1(ActionUpdater.java:93)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.callAction(ActionUpdater.java:130)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$new$2(ActionUpdater.java:94)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.update(ActionUpdater.java:427)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:290)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:270)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1707)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:270)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroup(ActionUpdater.java:181)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroup(ActionUpdater.java:155)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$expandActionGroupWithTimeout$11(ActionUpdater.java:197)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$3(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:163)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:585)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:150)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.withTimeout(ProgressIndicatorUtils.java:277)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroupWithTimeout(ActionUpdater.java:197)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroupWithTimeout(ActionUpdater.java:189)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:101)
	at com.intellij.openapi.actionSystem.impl.ActionPopupMenuImpl$MyMenu.show(ActionPopupMenuImpl.java:117)
	at com.intellij.ui.PopupHandler$2.invokePopup(PopupHandler.java:88)
	at com.intellij.ui.PopupHandler.mouseReleased(PopupHandler.java:48)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:298)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6651)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
	at com.intellij.ui.treeStructure.Tree.processMouseEvent(Tree.java:391)
	at java.desktop/java.awt.Component.processEvent(Component.java:6416)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5026)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2773)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:906)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:844)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:776)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:422)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:698)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:421)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by lstocchi** [comment]
> Also tried to test it but never got the Hub dialog being shown: I selected the Import from Hub on the Tasks node and got this exception being logged:


It should be fixed by https://github.com/redhat-developer/intellij-common/commit/ffacb558d488ae3354e8f5e9425c7ead09ae6278

**Comment by jeffmaury** [comment]
> 
> 
> > Also tried to test it but never got the Hub dialog being shown: I selected the Import from Hub on the Tasks node and got this exception being logged:
> 
> It should be fixed by [redhat-developer/intellij-common@ffacb55](https://github.com/redhat-developer/intellij-common/commit/ffacb558d488ae3354e8f5e9425c7ead09ae6278)

Then the PR is missing the switch to intellij-common 1.2.2

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**